### PR TITLE
python@3.9 3.9.1

### PR DESCRIPTION
**Pull request by carlocab**
Created with `brew bump-formula-pr`.

- version bumped to 3.9.1
- resource `setuptools` bumped to 51.0.0

Not sure if I'll manage to pull off this PR, but let's see.

### CODE REVIEW

**Comment by Rylan12** [comment]
Haha, just beat me to it!

**Comment by Rylan12** [comment]
The biggest thing off the top of my head is that there are some patches that may need to be removed. I haven't looked at the source to confirm, yet.

**Comment by carlocab** [comment]
Had a look at the patches; didn't seem like they needed to be removed based on the comments. Though perhaps I misunderstood/missed something.

**Comment by carlocab** [comment]
Oh, wait, I found it: https://github.com/python/cpython/commits/3.9?after=f0e42ae03c41ec32fcb3064772f46ff7f2c5ff3b+34&branch=3.9

Commits on 22 November. Looks like they've added Big Sur/ARM support, which means the first patch may need to be removed. Should I go ahead and do that, @Rylan12?

**Comment by Rylan12** [comment]
It looks to me like both patches can be removed. I know that FX's patch (the second one) from https://github.com/python/cpython/pull/23622 has been applied in https://github.com/python/cpython/commit/09a698b4743c669983d606595a1b2daeff6c3cf8.

The other one is longer so I haven't looked at it yet. Usually, though, the patch would fail to apply so we'd get a build error

**Comment by carlocab** [comment]
Yes, I see that commit in the history for 3.9 too. Removing the second patch.

**Comment by sharris40** [comment]
Just to note, `readline` is being updated simultaneously in https://github.com/Homebrew/homebrew-core/pull/66472, but I doubt there will be any ABI breakage.

**Comment by carlocab** [comment]
Think it's worth holding off on/rerunning the Python tests till/when `readline` is done?

**Comment by fxcoudert** [comment]
We have a bit of a backlog of formulas to retest, so I've killed testing. Let's redo it after `readline` is merged.

Also: my patch from https://github.com/python/cpython/pull/23556 was committed to the 3.9 branch, but did it make it into the 3.9.1 release? Can someone actually check that, please?

**Comment by carlocab** [comment]
@fxcoudert I believe it's the last commit (https://github.com/python/cpython/commit/09a698b4743c669983d606595a1b2daeff6c3cf8) listed on 3 December: https://github.com/python/cpython/commits/3.9?before=f0e42ae03c41ec32fcb3064772f46ff7f2c5ff3b+35&branch=3.9

**Comment by Rylan12** [comment]
@fxcoudert I'm pretty sure it is. Here's a direct link to a line modified by your commit on the 3.9.1 tag: https://github.com/python/cpython/blob/v3.9.1/Lib/distutils/spawn.py#L57-L58

**Comment by sharris40** [comment]
> We have a bit of a backlog of formulas to retest, so I've killed testing. Let's redo it after `readline` is merged.

More importantly, `openssl` also just updated upstream, so whenever somebody familiar with it makes a PR, this one should probably wait on that as well. Getting `openssl` patched is a bigger concern than Python.

**Comment by fxcoudert** [comment]
openssl 1.1.1i PR: https://github.com/Homebrew/homebrew-core/pull/66501

**Comment by carlocab** [comment]
**TL;DR** No new failures so far.

10.14:
```
Error: 16 failed steps!
brew test --retry --verbose cargo-edit
brew test --retry --verbose cucumber-cpp
brew fetch --retry diff-pdf
brew test --retry --verbose glyr
brew test --retry --verbose hyperkit
brew fetch --retry mapnik
brew install mariadb
brew install mariadb@10.1
brew install mariadb@10.2
brew install mariadb@10.3
brew install mariadb@10.4
brew test --retry --verbose mysql@5.6
brew test --retry --verbose mysql@5.7
brew test --retry --verbose pcl
brew install percona-server
brew test --retry --verbose pipgrip
```

10.15:
```
Error: 14 failed steps!
brew test --retry --verbose cargo-edit
brew fetch --retry diff-pdf
brew test --retry --verbose glyr
brew test --retry --verbose hyperkit
brew fetch --retry mapnik
brew install mariadb
brew install mariadb@10.1
brew install mariadb@10.2
brew install mariadb@10.3
brew install mariadb@10.4
brew test --retry --verbose mysql@5.6
brew test --retry --verbose mysql@5.7
brew test --retry --verbose pcl
brew install percona-server
```

**Comment by carlocab** [comment]
Some of these _might_ be new, I'll need to check.

11.0:
```
Error: 27 failed steps!
brew test --retry --verbose asuka
brew test --retry --verbose augustus
brew test --retry --verbose cargo-edit
brew test --retry --verbose corral
brew test --retry --verbose creduce
brew fetch --retry diff-pdf
brew install envoy
brew test --retry --verbose glyr
brew test --retry --verbose hyperkit
brew test --retry --verbose innotop
brew fetch --retry mapnik
brew install mariadb
brew install mariadb@10.1
brew install mariadb@10.2
brew install mariadb@10.3
brew install mariadb@10.4
brew test --retry --verbose mysql@5.6
brew test --retry --verbose mysql@5.7
brew test --retry --verbose networkit
brew install --only-dependencies oclgrind
brew install oclgrind
brew test --retry --verbose odin
brew test --retry --verbose pcl
brew install percona-server
brew test --retry --verbose pipgrip
brew test --retry --verbose pony-stable
brew test --retry --verbose ponyc
```
